File: all-submissions/0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length < 2){
            return 0;
        }
        int minVal = prices[0];
        int maxProfit = 0;
        for(int i=1; i< prices.length; i++){
            if (prices[i]< minVal){
                minVal = prices[i];
            } else if (prices[i] - minVal > maxProfit) {
                maxProfit = prices[i] - minVal;
            }
        }
        return maxProfit;
    }
}

File: all-submissions/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        buyPrice = float("inf")
        for i in range(len(prices)):
            if prices[i] <  buyPrice:
                buyPrice = prices[i]
            else:
                max_profit = max(max_profit, prices[i]-buyPrice)
        return max_profit
